Content.Server/Nyanotrasen/Explosion/Components/SelfExploderComponent.cs

```diff
@@ -0,0 +1,20 @@
+using Content.Shared.Actions.ActionTypes;
+using Content.Shared.Damage;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Server.Explosion.Components
+{
+    [RegisterComponent]
+    public sealed class SelfExploderComponent : Component
+    {
+        [DataField("explodeSelfAction", customTypeSerializer: typeof(PrototypeIdSerializer<InstantActionPrototype>))]
+        public string ExplodeSelfAction = "ExplodeSelf";
+
+        /// <summary>
+        /// Self damage, enough and of the correct type to trigger destruction.
+        /// </summary>
+        [DataField("selfDamage", required:true)]
+        [ViewVariables(VVAccess.ReadWrite)]
+        public DamageSpecifier SelfDamage = default!;
```

Comment:
Is `DestroyEntity` not sufficient? Or `DestroyedAt`?

https://github.com/Nyanotrasen/Nyanotrasen/blob/e136e46bed748e692f3fa7f3bfb6cb791c12d4c2/Content.Server/Nyanotrasen/GameTicking/Rules/ShipwreckedRuleSystem.cs#L911-L919

Reply:
DestroyEntity does not trigger destruction behaviors, it's what's called after all the destruction behaviors are finished.

That code block should work though, I'll try that